src/Components/Publishing/Sections/Sections.tsx

```diff
@@ -177,18 +181,94 @@ export class Sections extends Component<Props, State> {
     return sectionComponent
   }
 
+  getAdUnit(index: number) {
+    const { isMobile } = this.props
+
+    if (index === 6) {
+      return isMobile
+        ? AdUnit.Mobile_Feature_InContentLeaderboard1
+        : AdUnit.Desktop_Feature_Leaderboard1
+    }
+
+    if (index === 12) {
+      return isMobile
+        ? AdUnit.Mobile_Feature_InContentLeaderboard2
+        : AdUnit.Desktop_Feature_Leaderboard2
+    }
+
+    return isMobile
+      ? AdUnit.Mobile_Feature_InContentLeaderboard3
+      : AdUnit.Desktop_Feature_LeaderboardRepeat
+  }
+
+  /**
+   * Add a margin top when to the ad when the section is any of the following:
+   * 1) Text section without <br> tags wrapping the <p> or <h2> content
+   * 3) Image set section
+   * 4) Image collection with a caption/title
+   */
+  getAdMarginTop(section) {
+    const shouldAddTopMargin = () => {
+      switch (section.type) {
+        case "image_set":
+          return true
+        case "image_collection":
+          get(section, ["images", "[0]", "caption"])
+          return true
+        case "text":
+          section.body.includes("<br>")
+          return true
+        default:
+          return false
+      }
+    }
+
+    return shouldAddTopMargin() ? "calc(100% - 96%)" : null
+  }
```

Comment:
Is it safe to return a null margin? Perhaps return 0?

Reply:
I tried returning 0 but that had unintended consequences for the sections that didn't need the marging adjustment. Maybe auto will work?

Reply:
It's all good -- I see below that this is passed into a `styled-system` prop which should handle things fine.